lang/syn/src/codegen/accounts/constraints.rs

```diff
@@ -157,7 +157,7 @@ pub fn generate_constraint_zeroed(f: &Field, _c: &ConstraintZeroed) -> proc_macr
             __disc_bytes.copy_from_slice(&__data[..8]);
             let __discriminator = u64::from_le_bytes(__disc_bytes);
             if __discriminator != 0 {
-                return Err(anchor_lang::error::ErrorCode::ConstraintZero.into());
+                return anchor_lang::anchor_attribute_error::error!(anchor_lang::error::ErrorCode::ConstraintZero);
```

Comment:
What line will this give? I could be wrong, but I think we need to instead use the span information in the `Constraint` to get the correct line (which should be the line of the `#[account(zero)]` macro.